app/core/utils.coffee

```diff
@@ -162,6 +162,20 @@ orderedCourseIDs = [
   courseIDs.COMPUTER_SCIENCE_6
 ]
 
+courseNumericalStatus = {}
+courseNumericalStatus[courseIDs.COMPUTER_SCIENCE_6] = 1024
```

Comment:
what do these numbers signify?

Reply:
using bits to signify the student courses state.
you see we have cs1-6, gd1-3, wd1-2
then using bits:
`cs6,cs5,cs4,cs3,cs2,cs1,gd3,gd2,gd1,wd2,wd1`
`11111111111` to represent a student have access permission to all courses
`10000000000` to  represent a student only have access permission to cs6 (for example. not exactly in our server)

something like this

Reply:
as for teacher read/write permission. i'm not sure. i don't think it is used for teacher themselves. but maybe we can use that way.